src/h5web/nexus-visualizer/utils.ts

```diff
@@ -0,0 +1,32 @@
+import type { HDF5Link, HDF5Group } from '../providers/models';
+import { Vis } from '../dataset-visualizer/models';
+
+export function getSignalLink(group: HDF5Group): HDF5Link | undefined {
+  const signalAttribute = group.attributes?.find((a) => a.name === 'signal');
+
+  if (!signalAttribute) {
+    return undefined;
+  }
+
+  return group.links?.find((l) => l.title === signalAttribute.value);
+}
```

Comment:
As `getSignalLink` is written, it returns `undefined` if the `signal` attribute is not found (guard clause) **or** if the link does not exist (`group.links?.find`). This means both causes are undistinguishable for the consumer but I think it is not a big issue.